integration_test.go

```diff
@@ -0,0 +1,358 @@
+	"subfieldVarUInt16bit.ion",
+	"subfieldVarUInt32bit.ion",
+	"symbolEmpty.ion",
+	"symbols.ion",
+	"testfile22.ion",
+	"testfile23.ion",
+	"testfile31.ion",
+	"testfile35.ion",
+	"testfile37.ion",
+	"utf16.ion",
+	"utf32.ion",
+	"whitespace.ion",
+}
+
+var textRoundTripSkipList = []string{
+	"allNulls.ion",
+	"annotations.ion",
+	"bigInts.ion",
+	"clobWithNonAsciiCharacter.10n",
+	"clobs.ion",
+	"decimal64BitBoundary.ion",
+	"decimal_values.ion",
+	"decimals.ion",
+	"decimalsWithUnderscores.ion",
+	"float_zeros.ion",
+	"floats.ion",
+	"floatsVsDecimals.ion",
+	"intBigSize1201.10n",
+	"intBigSize13.10n",
+	"intBigSize14.10n",
+	"intBigSize16.10n",
+	"intBigSize256.10n",
+	"intBigSize256.ion",
+	"intBigSize512.ion",
+	"intLongMaxValuePlusOne.10n",
+	"localSymbolTableImportZeroMaxId.ion",
+	"notVersionMarkers.ion",
+	"nullDecimal.10n",
+	"nulls.ion",
+	"structWhitespace.ion",
+	"subfieldInt.ion",
+	"subfieldUInt.ion",
+	"subfieldVarInt.ion",
+	"subfieldVarUInt.ion",
+	"subfieldVarUInt15bit.ion",
+	"subfieldVarUInt16bit.ion",
+	"subfieldVarUInt32bit.ion",
+	"symbolEmpty.ion",
+	"symbols.ion",
+	"symbols.ion",
+	"symbols.ion",
+	"systemSymbols.ion",
+	"systemSymbolsAsAnnotations.ion",
+	"testfile22.ion",
+	"testfile23.ion",
+	"testfile24.ion",
+	"testfile31.ion",
+	"testfile35.ion",
+	"testfile37.ion",
+	"utf16.ion",
+	"utf32.ion",
+	"whitespace.ion",
+	"zeroFloats.ion",
+}
+
+func TestBinaryRoundTrip(t *testing.T) {
+	readFilesAndTest(t, goodPath, binaryRoundTripSkipList, func(t *testing.T, path string) {
+		binaryRoundTrip(t, path)
+	})
+}
+
+func TestTextRoundTrip(t *testing.T) {
+	readFilesAndTest(t, goodPath, textRoundTripSkipList, func(t *testing.T, path string) {
+		textRoundTrip(t, path)
+	})
+}
+
+func binaryRoundTrip(t *testing.T, fp string) {
+	b := loadFile(t, fp)
+
+	// Make a binary writer from the file
+	br := NewReaderBytes(b)
+	buf := bytes.Buffer{}
+	bw := NewBinaryWriter(&buf)
+	writeToWriterFromReader(t, br, bw)
+	bw.Finish()
+
+	// Make a text writer from the binary writer
+	r := NewReaderBytes(buf.Bytes())
+	str := strings.Builder{}
+	w := NewTextWriter(&str)
+	writeToWriterFromReader(t, r, w)
+	w.Finish()
+
+	// Make another binary writer using the text writer
+	br2 := NewReaderStr(str.String())
+	buf2 := bytes.Buffer{}
+	bw2 := NewBinaryWriter(&buf2)
+	writeToWriterFromReader(t, br2, bw2)
+	bw2.Finish()
+
+	// Compare the 2 binary writers
+	if !reflect.DeepEqual(bw, bw2) {
+		t.Errorf("Round trip test failed on: " + fp)
+	}
+}
+
+func textRoundTrip(t *testing.T, fp string) {
+	b := loadFile(t, fp)
+
+	// Make a binary writer from the file
+	r := NewReaderBytes(b)
+	str := strings.Builder{}
+	w := NewTextWriter(&str)
+	writeToWriterFromReader(t, r, w)
+	w.Finish()
+
+	// Make a text writer from the binary writer
+	br := NewReaderStr(str.String())
+	buf := bytes.Buffer{}
+	bw := NewBinaryWriter(&buf)
+	writeToWriterFromReader(t, br, bw)
+	bw.Finish()
+
+	// Make another binary writer using the text writer
```

Comment:
Please double check comments in this section.

Reply:
Lazy copy paste issue. Fixed, thanks.